pkg/controller/csi_handler_test.go

```diff
@@ -259,8 +270,12 @@ func TestCSIHandler(t *testing.T) {
 			addedVA:        va(false /*attached*/, "" /*finalizer*/, nil /* annotations */),
 			expectedActions: []core.Action{
 				// Finalizer is saved first
-				core.NewUpdateAction(vaGroupResourceVersion, metav1.NamespaceNone, va(false /*attached*/, fin, ann)),
-				core.NewUpdateAction(vaGroupResourceVersion, metav1.NamespaceNone, va(true /*attached*/, fin, ann)),
+				core.NewPatchAction(vaGroupResourceVersion, metav1.NamespaceNone, testPVName+"-"+testNodeName,
```

Comment:
I wish there was a way to pass only the new VA - the code that compares the expected actions with reality *could* get old VA from reactor, get patch from API call and compute new VA. I am not sure how hard would it be and if it's even feasible. Would you please check this possibility? Don't spend too much time on it, this PR is mostly OK even without it.

Reply:
Thanks, I review the codes, this may be possible. Since I do not have too much time, I would like to send anther PR to do this later. Do you think it is ok? 

Other two places you advised have changed.